appmap/test/data/pytest/Dockerfile

```
@@ -0,0 +1,7 @@
+FROM python:3.9
```

```suggestion
FROM python:3.9-alpine
```

Size difference: 332.72 MB vs 16.44MB

---

Altough might need to check dependency compatibility

---

Did a quick check. No longer applicable. 
`pip install poetry` has a dependency on `cryptography` which recently added dependency on `rust` which itself is around 392 MB.

https://github.com/pyca/cryptography/issues/5771

---

@apotterri I am seeing this error with `python:3.9`

```
=========================================================================================================== ERRORS ===========================================================================================================
_______________________________________________________________________________________________ ERROR collecting test session ________________________________________________________________________________________________
/usr/local/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1030: in _gcd_import
    ???
<frozen importlib._bootstrap>:1007: in _find_and_load
    ???
<frozen importlib._bootstrap>:986: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:680: in _load_unlocked
    ???
../../../_implementation/recording.py:227: in wrapped_exec_module
    exec_module(*args, **kwargs)
../../../_implementation/recording.py:227: in wrapped_exec_module
    exec_module(*args, **kwargs)
/usr/local/lib/python3.9/site-packages/_pytest/assertion/rewrite.py:170: in exec_module
    exec(co, module.__dict__)
tmp/pip-install-53_gxics/cryptography_5ab623b84cd5460694d6723754acd560/tests/conftest.py:10: in <module>
    from .utils import check_backend_support
../../../_implementation/recording.py:227: in wrapped_exec_module
    exec_module(*args, **kwargs)
tmp/pip-install-53_gxics/cryptography_5ab623b84cd5460694d6723754acd560/tests/utils.py:18: in <module>
    import cryptography_vectors
E   ModuleNotFoundError: No module named 'cryptography_vectors'
```

---

`pip install cryptography_vectors`

---

Ugh. I'm losing patience with poetry. It's getting in the way more than it's being helpful. I've been considering switch to just using pip, this might be the final straw.

@dividedmind @virajkanwade What do you guys think of switching to pip?

---

Well, looking around for alternatives, maybe poetry is actually the best choice. it won't be required for users `appmap-python`, so the size increase from requiring rust problably isn't such a big deal.

---

I don't think it's a good idea to switch to alpine. The base image is smaller but you end up having to download and compile a lot, so it makes it actually worse. (I know, I went down this rabbit hole.) See https://pythonspeed.com/articles/alpine-docker-python/ for a good write-up on the subject.

It might be worth it if you're creating some specific, limited, user-downloadable, slimmed down image. But it's not what we're doing here.

---

@dividedmind  Yes, alpine, in this case turns out to be a bigger overhead. As mentioned above I gave it a try and ended up downloading almost 600-700 MB packages.

---

I have it working with `FROM python:3.9.1-slim-buster` with very minimal installs